src/vs/editor/browser/controller/mouseHandler.ts

```diff
@@ -346,8 +346,10 @@ class MouseDownOperation extends Disposable {
 
 	private _mouseMoveMonitor: GlobalEditorMouseMoveMonitor;
 	private _mouseDownThenMoveEventHandler: EventGateKeeper<EditorMouseEvent>;
+	private _mouseDragThenMoveEventHandler: EventGateKeeper<EditorMouseEvent>;
```

Comment:
No need for a separate handler, I suggest to use `mouseState.lastMouseDownEvent` (which should be renamed to something better, see other comment) to differentiate the two cases inside `onMouseDownThenMove`